### PR TITLE
feature: Add field filtering to Loki logs

### PR DESCRIPTION
**Pull request by BravoNatalie**
### Description

This PR introduces a `logDataTransformer` callback option to the Loki Logging class, allowing users to filter or transform fields in a log.

### Motivation

This package is used in several workers, and the logs stored contain many unnecessary fields that can be removed to reduce log data size and lower costs. More details on this issue can be found at https://github.com/storacha/project-tracking/issues/175.

### Changes

- Add a `logDataTransformer` callback to the Loki Logging constructor for custom log transformation.
- Update `_add` method to apply `logDataTransformer ` when provided.
- Add unit tests for `loki.js`.
- Update type definitions for improved accuracy.

### CODE REVIEW

File: packages/utils/test/loki.test.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+import test from 'ava'
```
Comment (travis) [inline packages/utils/test/loki.test.js:1]: thank you for adding tests for this!

File: packages/utils/src/loki.js [hunk 2]
```diff
@@ -225,12 +241,13 @@ export class Logging {
...
 
   /**
-   * Add log entry to batch
+   * Add log entry to batch after applying field filtering
    *
    * @param {any} body
    */
   _add(body) {
-    this.logEventsBatch.push(body)
+    const log = this.filterFields ? this.filterFields(body) : body
```
Comment (travis) [inline packages/utils/src/loki.js:249]: I think this is the right idea, but I'm getting a bit hung up on the name `filterFields` - this callback could really be doing any sort of transformation here and it feels like the name should probably reflect that - maybe `logMessageTransformer` or something? It's a bit wordy, but it only gets written out when the `Logging` object gets set up so shouldn't have a huge practical impact on DX. Open to other names if you've got ideas!

File: packages/utils/test/loki.test.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  t.is(log._times.get('testing').name, 'testing')
+  t.is(log._times.get('testing').description, 'description')
+  t.assert(Date.now() >= log._times.get('testing').start)
+})
+
+test('should not log with timeend', async (t) => {
+  const log = logging()
+
+  log.time('testing')
+  await sleep(2000)
```
Comment (travis) [inline packages/utils/test/loki.test.js:71]: does this need sleep for this long? it would be helpful to have a comment here explaining why the sleep is necessary, and I'd dial it down to 100-500 ms if feasible

**Review by travis — CHANGES_REQUESTED** [review]
This looks great in general! Need to sort out the name of this callback before we can merge, and there's a test improvement that would be a nice-to-have, but this is looking good.

File: packages/utils/src/loki.js (hunk 2 above)
Comment (BravoNatalie) [reply packages/utils/src/loki.js:249]: Thanks for the heads-up! I agree—it makes sense to clarify that this is a transformer. I suggest `logDataTransformer` to help differentiate it from the message field in the log object.

File: packages/utils/test/loki.test.js (hunk 3 above)
Comment (BravoNatalie) [reply packages/utils/test/loki.test.js:71]: Sure, I'll reduce it to the minimum possible for easier differentiation in the test.

File: packages/utils/src/loki.js [hunk 4]
```diff
@@ -225,12 +241,13 @@ export class Logging {
   }
 
   /**
-   * Add log entry to batch
+   * Add log entry to batch after applying field filtering
```
Comment (travis) [inline packages/utils/src/loki.js:244]:
```suggestion
   * Add log entry to batch after applying logDataTransformer

**Review by travis — APPROVED** [review]
Looks good! I have one suggested change in a doc comment but otherwise this looks great.
